octoprint_octorelay/__init__.py

```diff
@@ -171,8 +171,9 @@ def on_event(self, event, payload):
             self.handle_plugin_event(PRINTING_STOPPED)
         elif hasattr(Events, "CONNECTIONS_AUTOREFRESHED"): # Requires OctoPrint 1.9+
             if event == Events.CONNECTIONS_AUTOREFRESHED:
-                self._logger.debug("Connecting to the printer")
-                self._printer.connect()
+                if payload is not None and "ports" in payload and len(payload["ports"]) > 0:
```

Comment:
this prevents:
- redundant attempts to connect when this event being fired initially without ports
- attempts to connect after disconnecting (the event is also fired)